### ISSUE TITLE
Unable to run 1.4.7 repack on postgres 10.20

### ISSUE THREAD

**Issue by zoonage** [reorg/pg_repack#307]
```
root@7eff292cde6d:/# pg_repack --no-superuser-check -T 10000 -t app_install_tokens --elevel=DEBUG
DEBUG: No workers to disconnect.
INFO: repacking table "public.app_install_tokens"
DEBUG: ---- repack_one_table ----
DEBUG: target_name       : public.app_install_tokens
DEBUG: target_oid        : 499505472
DEBUG: target_toast      : 1849957401
DEBUG: target_tidx       : 1849957403
DEBUG: pkid              : 499505480
DEBUG: ckid              : 0
DEBUG: create_pktype     : SELECT repack.create_index_type(499505480,499505472)
DEBUG: create_log        : SELECT repack.create_log_table(499505472)
DEBUG: create_trigger    : CREATE TRIGGER repack_trigger AFTER INSERT OR DELETE OR UPDATE ON public.app_install_tokens FOR EACH ROW EXECUTE PROCEDURE repack.repack_trigger(499505472,'id')
DEBUG: enable_trigger    : ALTER TABLE public.app_install_tokens ENABLE ALWAYS TRIGGER repack_trigger
DEBUG: create_table      : SELECT repack.create_table_table(499505472,'pg_default') --  WITH NO DATA
DEBUG: copy_data         : INSERT INTO repack.table_499505472 SELECT id,account_id,customer_id,NULL::integer AS "........pg.dropped.4........",NULL::integer AS "........pg.dropped.5........",dt_updated,app_install_id,token,om_app_id,platform,notifications_enabled,dt_notifications_last_enabled,dt_notifications_last_disabled,dt_notifications_first_enabled,dt_notifications_first_disabled FROM ONLY public.app_install_tokens
DEBUG: alter_col_storage : (skipped)
DEBUG: drop_columns      : ALTER TABLE repack.table_499505472 DROP COLUMN "........pg.dropped.4........", DROP COLUMN "........pg.dropped.5........"
DEBUG: delete_log        : DELETE FROM repack.log_499505472
DEBUG: lock_table        : LOCK TABLE public.app_install_tokens IN ACCESS EXCLUSIVE MODE
DEBUG: sql_peek          : SELECT * FROM repack.log_499505472 ORDER BY id LIMIT $1
DEBUG: sql_insert        : INSERT INTO repack.table_499505472 VALUES ($1.*)
DEBUG: sql_delete        : DELETE FROM repack.table_499505472 WHERE (id) = ($[1.id](http://1.id/))
DEBUG: sql_update        : UPDATE repack.table_499505472 SET (id, account_id, customer_id, dt_updated, app_install_id, token, om_app_id, platform, notifications_enabled, dt_notifications_last_enabled, dt_notifications_last_disabled, dt_notifications_first_enabled, dt_notifications_first_disabled) = ($[2.id](http://2.id/), $2.account_id, $2.customer_id, $2.dt_updated, $2.app_install_id, $2.token, $2.om_app_id, $2.platform, $2.notifications_enabled, $2.dt_notifications_last_enabled, $2.dt_notifications_last_disabled, $2.dt_notifications_first_enabled, $2.dt_notifications_first_disabled) WHERE (id) = ($[1.id](http://1.id/))
DEBUG: sql_pop           : DELETE FROM repack.log_499505472 WHERE id IN (
DEBUG: ---- setup ----
DEBUG: index[0].target_oid      : 499505480
DEBUG: index[0].create_index    : SELECT repack.create_index(499505480, 499505472, NULL)
DEBUG: index[1].target_oid      : 499505482
DEBUG: index[1].create_index    : SELECT repack.create_index(499505482, 499505472, NULL)
DEBUG: index[2].target_oid      : 1058088952
DEBUG: index[2].create_index    : SELECT repack.create_index(1058088952, 499505472, NULL)
DEBUG: index[3].target_oid      : 1097233184
DEBUG: index[3].create_index    : SELECT repack.create_index(1097233184, 499505472, NULL)
DEBUG: LOCK TABLE public.app_install_tokens IN ACCESS SHARE MODE
DEBUG: No competing DDL to cancel.
DEBUG: Waiting on ACCESS SHARE lock...
DEBUG: ---- copy tuples ----
DEBUG: No competing DDL to cancel.
ERROR: query failed: ERROR:  must be owner of relation table_499505472
DETAIL: query was: ALTER TABLE repack.table_499505472 DROP COLUMN "........pg.dropped.4........", DROP COLUMN "........pg.dropped.5........"
DEBUG: No workers to disconnect.
NOTICE:  table "table_499505472" does not exist, skipping
root@7eff292cde6d:/#
```

The schema of this table is

```
ometria_ecomm_0004=> \d app_install_tokens
Table "public.app_install_tokens"
Column | Type | Modifiers
---------------------------------+--------------------------+-----------------------------------------------------------------
id | integer | not null default nextval('app_install_tokens_id_seq'::regclass)
account_id | integer | not null
customer_id | integer | not null
dt_updated | timestamp with time zone | default now()
app_install_id | character varying(255) | not null
token | character varying(255) | not null
om_app_id | character varying(255) |
platform | character varying(255) |
notifications_enabled | boolean |
dt_notifications_last_enabled | timestamp with time zone |
dt_notifications_last_disabled | timestamp with time zone |
dt_notifications_first_enabled | timestamp with time zone |
dt_notifications_first_disabled | timestamp with time zone |
Indexes:
"app_install_tokens_pkey" PRIMARY KEY, btree (id)
"idx_app_install_tokens_account_id_customer_id_app_install_id_pl" UNIQUE, btree (account_id, customer_id, app_install_id, platform)
"idx_app_install_tokens_account_id_customer_id_token_platform" UNIQUE, btree (account_id, customer_id, token, platform)
"idx_app_install_tokens_account_id_customer_id" btree (account_id, customer_id)
Foreign-key constraints:
"fk_customer" FOREIGN KEY (customer_id) REFERENCES customers(id) ON DELETE CASCADE
```

This does work on other tables that I've done as a spot check

**Comment by zoonage** [comment]
@Melkij  this works on other tables in the DB, just specifically not this table. The user I'm running as (`avnadmin`) has the role that owns the table granted to it

**Comment by evenfrost** [comment]
@zoonage have you manage to handle this? Faced the same issue when running `pg_repack` in DigitalOcean's Managed Databases, which doesn't provide superuser access. Though with the `-k` option some tables are being repacking without issues, but for some I'm getting this error:
```
ERROR: query failed: ERROR:  must be owner of table table_527343
```
on the `drop_columns` operations:
```
DEBUG: drop_columns      : ALTER TABLE repack.table_527343 DROP COLUMN "........pg.dropped.13........"
```

For the tables that are working this operation is just being skipped though:
```
DEBUG: drop_columns      : (skipped)
```

**Comment by zoonage** [comment]
@evenfrost We've not found a solution yet unfortunately

**Comment by mkherlakian** [comment]
@zoonage I did some digging into this as we are also hosted on Aiven. I have an explanation, but not a solution (yet). Hoping that others may help with a workaround.

The reason why this works for some tables and not others is dropped columns.

If at some point in your table's (the one you're repacking)'s life you dropped one or more columns, those still exist in the underlying schema, and repack will insert null values during its copy data phase, and will subsequently drop the column. 

Those are the columns named like so `........pg.dropped.4.......` - If you look at your table's history, at some point you dropped 2 columns between `customer_id` and `dt_updated`...

Now the problem is that while you accurately described that your original table is owned by avnadmin, and the `table_<oid>` that gets created is also owned by the same user, that's not the case at table creation. instead, when pg_repack executed the function that creates the table (create_table_table provided by the pg_repack extension), the table is created with owner `postgres`.

And in postgressql, only the table's owner is allowed to issue `ALTER TABLE` statements against that table. Therefore by the time 
https://github.com/reorg/pg_repack/blob/9e6c048801d1c59fc6ab843a2fd025eb41c38c63/bin/pg_repack.c#L1530 

runs, the table is still owned by postgres - it's only later when `repack_swap` is run that ownership changes as I understand it. 

If the alter table statement was run through a function provided by pg_repack this wouldn't be an issue (it would actually be a fix I believe?).

I filed a ticket with Aiven as well. I'll update when I hear back.
